tools/cli/src/mentat_cli/repl.rs

```diff
@@ -7,63 +7,100 @@
+        let mut map = HashMap::new();
+        map.insert(HELP_COMMAND, "Show help for commands.");
+        map.insert(OPEN_COMMAND, "Open a database at path.");
+        map
+    };
+}
 
 /// Executes input and maintains state of persistent items.
 pub struct Repl {
+    store: Store,
 }
 
 impl Repl {
     /// Constructs a new `Repl`.
-    pub fn new() -> Repl {
-        Repl{}
+    pub fn new(db_name: Option<String>) -> Repl {
+        Repl{
+            store: Store::new(db_name),
+        }
     }
 
-
     /// Runs the REPL interactively.
     pub fn run(&mut self) {
-        let mut more = false;
+        let mut more: Option<Command> = None;
         let mut input = InputReader::new();
 
         loop {
-            let res = input.read_input(if more { MORE_PROMPT } else { DEFAULT_PROMPT });
-        //     let res = if self.read_block {
-        //         self.read_block = false;
-        //         input.read_block_input(BLOCK_PROMPT)
-        //     } else {
-        //         input.read_input(if more { MORE_PROMPT } else { DEFAULT_PROMPT })
-        //     };
+            let res = input.read_input(if more.is_some() { MORE_PROMPT } else { DEFAULT_PROMPT });
 
             match res {
-                Command(name, args) => {
-                    debug!("read command: {} {:?}", name, args);
-
-                    more = false;
-                    self.handle_command(name, args);
+                MetaCommand(cmd) => {
+                    debug!("read command: {:?}", cmd);
+                    more = None;
+                    self.handle_command(cmd);
                 },
                 Empty => (),
-                More => { more = true; },
+                More(cmd) => { more = Some(cmd); },
                 Eof => {
                     if input.is_tty() {
                         println!("");
                     }
                     break;
-                }
+                },
+                InputError(err) => {
+                    println!("{}", err);
+                    more = None;
+                },
             };
         }
     }
 
     /// Runs a single command input.
-    fn handle_command(&mut self, cmd: String, args: Option<String>) {
-        println!("{:?} {:?}", cmd, args);
+    fn handle_command(&mut self, cmd: Command) {
+        match cmd {
+            Command::Help(args) => self.help_command(args),
+            Command::Open(db) => {
+                self.store.open(Some(db));
+            },
+            Command::Close => self.store.close(),
+            Command::Err(message) => println!("{}", message),
+            _ => unimplemented!(),
+        }
+    }
+
+    fn help_command(&self, args: Vec<String>) {
+        if args.is_empty() {
+            for (cmd, msg) in COMMAND_HELP.iter() {
+                println!(".{} - {}", cmd, msg);
+            }
+        } else {
+            for arg in args {
+                let msg = COMMAND_HELP.get(arg.as_str());
```

Comment:
You should allow the command arg to start with a `.` — I immediately typed `.help .open` and got all of the commands, 'cos it wanted `.help open`.